### PR TITLE
web: seal CSS modules migration with linter rules

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

The CSS modules migration [was completed](https://github.com/sourcegraph/sourcegraph/issues/23015) at the beginning of the month, but we didn't have time to seal it with linter rules that would allow closing the issue. This PR addresses that.

Screen Shot 2022-01-28 at 17 31 01

👆10 Global CSS files left are not meant for migration because they introduce global styles not tied to any specific React component.

## Changes

- Added new package `stylelint-plugin-stylelint` to the `client` folder.
- Added one custom stylelint rule `filenames/match-regex`.
- Enabled this rule to **ban non-module SCSS files outside of designated folders**.

## Notes 

- This PR should be merged after updating our stylelint-config: https://github.com/sourcegraph/stylelint-config/pull/153
- More context about the migration can be found in [the CSS isolation RFC](https://docs.google.com/document/d/1VedPoFy7WIuHeXPzwRGMmgjDPL4CBdTKmhy4OBmlmf8/edit#heading=h.trqab8y0kufp).
- The Code Insight lives on the [Frontend Platform migrations dashboard](https://k8s.sgdev.org/insights/dashboards/ZGFzaGJvYXJkOnsiSWRUeXBlIjoiY3VzdG9tIiwiQXJnIjo3MjcxNDZ9).

Closes the tracking issue https://github.com/sourcegraph/sourcegraph/issues/23010. 🤩

### CODE REVIEW

File: client/stylelint-plugin-sourcegraph/lib/rules/filenames-match-regex.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const regexp = new RegExp(secondaryOptionObject)
+
+  return function (postcssRoot, postcssResult) {
+    const filename = postcssRoot.source.input.from
+    const name = path.basename(filename)
+    const matchRegex = regexp.test(name)
+
+    if (!matchRegex) {
+      stylelint.utils.report({
+        message: `Filename ${name} does not match the regex expression.`,
```
Comment (eseliger) [inline client/stylelint-plugin-sourcegraph/lib/rules/filenames-match-regex.js:22]:
```suggestion
        message: `Filename ${name} does not match the regular expression.`,
```

**Review by eseliger — APPROVED** [review]

File: client/stylelint-plugin-sourcegraph/lib/rules/filenames-match-regex.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// @ts-check
+
+const stylelint = require('stylelint')
+const path = require('path')
+
+const ruleName = 'filenames/match-regex'
+const messages = stylelint.utils.ruleMessages(ruleName, {
+  expected: 'Yo expected ...',
```
Comment (umpox) [inline client/stylelint-plugin-sourcegraph/lib/rules/filenames-match-regex.js:8]: What does this do? Is it meant to say "yo expected"? 😎

File: client/stylelint-plugin-sourcegraph/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# stylelint-plugin-sourcegraph
```
Comment (valerybugakov) [inline client/stylelint-plugin-sourcegraph/README.md:1]: Later we can migrate this plugin to the codemod repo to re-use codemods logic: https://github.com/sourcegraph/codemod/tree/main/packages if needed.

**Review by umpox — APPROVED** [review]
Going to be amazing to close that tracking issue 🙌

File: client/stylelint-plugin-sourcegraph/lib/rules/filenames-match-regex.js (hunk 2 above)
Comment (valerybugakov) [reply client/stylelint-plugin-sourcegraph/lib/rules/filenames-match-regex.js:8]: `Yo` should not be there. But everything else is just a boilerplate from [stylelint docs](https://stylelint.io/developer-guide/plugins#the-anatomy-of-a-plugin).

**Pull request merged**